src/paragraph.js

```diff
@@ -38,5 +49,174 @@ export default class Paragraph extends Plugin {
 		buildViewConverter().for( data.viewToModel )
 			.fromElement( 'p' )
 			.toElement( 'paragraph' );
+
+		// Autoparagraph text.
+		data.viewToModel.on( 'text', ( evt, data, consumable, conversionApi ) => {
+			autoparagraphText( doc, evt, data, consumable, conversionApi );
+		}, { priority: 'lowest' } );
+
+		// Post-fix potential subsequent paragraphs created by autoparagraphText().
+		data.viewToModel.on( 'element', mergeSubsequentParagraphs, { priority: 'lowest' } );
+		data.viewToModel.on( 'documentFragment', mergeSubsequentParagraphs, { priority: 'lowest' } );
+
+		// Convert paragraph-like elements to paragraphs if they weren't consumed.
+		// It's a 'low' priority in order to hook in before the default 'element' converter
+		// which would then convert children before handling this element.
+		data.viewToModel.on( 'element', ( evt, data, consumable, conversionApi ) => {
```

Comment:
But these converters can be split into two groups – those for the text autop and generic block conv. The 3 above ones handle text. The one below is a generic block conv. I think this is a good order in this case.